tests/test_vms_platform.py

```diff
@@ -21,177 +19,171 @@
-            self._validate_vm_platform(vm)
-
-    def test_dispvm_default_platform(self):
-        """
-        Query dom0 Qubes preferences and confirm that new DispVMs
-        will be created under a supported OS. Requires a separate
-        test because DispVMs may not be running at present.
-        """
-        cmd = ["qubes-prefs", "default_dispvm"]
-        result = subprocess.check_output(cmd).decode("utf-8").rstrip("\n")
-        assert result == "sd-viewer"
-
-    def test_sd_vm_apt_sources(self):
-        """
-        Test that the three templates we install our apt sources into are correct
-        """
-        for vm_name in [
-            SD_TEMPLATE_SMALL,
-            SD_TEMPLATE_LARGE,
-        ]:
-            vm = self.app.domains[vm_name]
-
-            # First, check the prod apt repo is configured, which happens unconditionally
-            # via securedrop-keyring deb package
-            self.assert_apt_source(
-                vm,
-                "main",
-                "https://apt.freedom.press",
-                "/etc/apt/sources.list.d/apt_freedom_press.sources",
-            )
-
-            if self.config["environment"] == "prod":
-                # No test sources should be present
-                stdout, stderr = vm.run("ls /etc/apt/sources.list.d/")
-                sources_list = stdout.decode("utf-8").rstrip("\n")
-                assert "apt-test_freedom_press.sources" not in sources_list
-                continue
-
-            # we're in staging or dev, so check for that file
-            components = ["main"]
-            if self.config["environment"] == "dev":
-                components.append("nightlies")
-
-            self.assert_apt_source(
-                vm,
-                " ".join(components),
-                "https://apt-test.freedom.press",
-                "/etc/apt/sources.list.d/apt-test_freedom_press.sources",
-            )
-
-    def assert_apt_source(self, vm, component, url, filename):
-        stdout, stderr = vm.run(f"cat {filename}")
-        contents = stdout.decode("utf-8").rstrip("\n")
-
-        assert f"Components: {component}\n" in contents, f"{vm.name} wrong component"
-        assert f"URIs: {url}\n" in contents, f"{vm.name} wrong URL"
-        assert (
-            f"Suites: {SUPPORTED_SD_DEBIAN_DIST}\n" in contents
-        ), f"{vm.name} wrong suite/codename"
+        except subprocess.CalledProcessError:
+            pytest.fail(fail_msg)
+        # 'stdout' will contain timestamped progress info; ignore it
+        results = stderr.rstrip().decode("utf-8")
+        stdout_lines = results.split("\n")
+        # filter out gpgcheck warning present on all dnf operations
+        stdout_lines = [
+            x for x in stdout_lines if not x.startswith("Warning: Enforcing GPG signature check")
+        ]
+        results = "".join(stdout_lines)
+        assert results == "", fail_msg
+
+
+@unittest.skipIf(IS_CI, "Skipping on CI")
```

Comment:
You can use https://docs.pytest.org/en/6.2.x/skipping.html#id1

Reply:
I had originally considering including these, but I ultimately decided to focus on just what was on the critical path for #1373. There are many other things I'd want address, but I had to inhibit myself from doing so, otherwise this would have been a massive PR. I only included [this scope-creep suggestion](https://github.com/freedomofpress/securedrop-workstation/pull/1487#discussion_r2540931203) because it was clearly an issue with the test and would lead to conflicts (due to indentation changes) if done in another PR.

So if you wouldn't mind, I'd prefer to address this separately. I have added this as a note in https://github.com/freedomofpress/securedrop-workstation/pull/1487#discussion_r2539418618, so we don't forget.

Reply:
Sounds good.